### ISSUE TITLE
Steam sdk support? e.g. fxgl-steam module

### ISSUE THREAD

**Issue by chengenzhao** [AlmasB/FXGL#1002]
**A clear and concise description of what the feature is.**
To provide Java module version(should be better in jmod format) of Steamworks SDK
https://partner.steamgames.com/doc/sdk

**Describe alternatives you've considered (optional)**
Steamworks4j 
https://github.com/code-disaster/steamworks4j
but it is not modularized thus we couldn't use javafx:jlink to build the custom runtime 

**Additional context (optional)**
If there is modularized jni wrapper of Steamworks SDK, then I could build custom FXGL runtime by using Gluon javafx plugin
<img width="1280" alt="Screen Shot 2021-01-20 at 8 31 24 PM" src="https://user-images.githubusercontent.com/5525436/105175143-7e5cda80-5b5e-11eb-83f2-9ab0bc6a766f.png">

**Comment by AlmasB** [comment]
This could be an interesting area to explore, however making the lib above (https://github.com/code-disaster/steamworks4j) modular is probably significantly easier rather than writing a new one.

**Comment by chengenzhao** [comment]
> This could be an interesting area to explore, however making the lib above (https://github.com/code-disaster/steamworks4j) modular is probably significantly easier rather than writing a new one.

indeed, but that project seems like deprecated already and the last commit is stay in about 2 years ago...

**Comment by chengenzhao** [comment]
I made it
here is modularized jars that could be used as jlink resources
the module name is steam.sdk
should used as 

`jlink --module-path ../jmods --add-modules steam.sdk,and other modules --output runtime`

[steam-sdk-linux-x64.zip](https://github.com/AlmasB/FXGL/files/6311332/steam-sdk-linux-x64.zip)
[steam-sdk-mac-x64.zip](https://github.com/AlmasB/FXGL/files/6311334/steam-sdk-mac-x64.zip)
[steam-sdk-win-x64.zip](https://github.com/AlmasB/FXGL/files/6311335/steam-sdk-win-x64.zip)

in zip file because github doesn't allow jar files to be uploaded
users could download zip files then rename the suffix to jar

I have tested in my macbook pro and it works pretty good with gluon javafx plugin
could use javafx:jlink to generate game image

cheers

**Comment by chengenzhao** [comment]
source code
in maven format
[steam-sdk.zip](https://github.com/AlmasB/FXGL/files/6311405/steam-sdk.zip)

**Comment by chengenzhao** [comment]
current my game module-info
too sad netty and vert.x is not manually modularized 
so I have to remove dependencies of vert.x
<img width="387" alt="Screen Shot 2021-04-14 at 10 16 15 PM" src="https://user-images.githubusercontent.com/5525436/114725422-0f62dc80-9d6f-11eb-9113-8985ba883aa3.png">

**Comment by AlmasB** [comment]
https://github.com/code-disaster/steamworks4j is under MIT license, so we could technically use your solution related to adding a module-info.java and integrate it into FXGL natively. However, given it already works for your use case, we can probably park this issue for now but keep it open for any future development.
